### PR TITLE
Fix TestMachinesNICs.{testNICDelete,testVmNICs} and improve debugging

### PR DESCRIPTION
**Pull request by martinpitt**
Our current top flake, with > 40% failure rate on rhel-9-4. Yet, doesn't immediately reproduce locally, so let's get some bot help.

image

### CODE REVIEW

**Comment by martinpitt** [comment]
Locally, when the test passes: After confirming the removal, this happens:
```
> debug: subVmTest1.GET_VM(/org/libvirt/QEMU/domain/_30100555_923e_4732_aae2_e776eb86d7c9, system): update props {"inactiveXML":{"connectionName":"system","name":"subVmTest1","id":"/org/libvirt/QEMU/domain/_30100555_923e_4732_aae2_e776eb86d7c9","osType":null,"osBoot":[{"order":1,"type":"disk"},{"order":2,"type":"network"}],"firmware":null,"arch":"x86_64","currentMemory":131072,"memory":131072,"memoryBacking":false,"vcpus":{"count":"1","placement":"static","max":"1"},"disks":{"vda":{"target":"vda","driver":{"name":"qemu","type":"qcow2","cache":null,"discard":null,"io":null,"errorPolicy":null},"type":"file","device":"disk","source":{"file":"/var/lib/libvirt/images/subVmTest1-2.img","dev":null,"pool":null,"volume":null,"protocol":null,"name":null,"host":{},"startupPolicy":null},"bus":"virtio","readonly":false,"shareable":false,"removable":null}},"emulatedMachine":"pc-i440fx-rhel7.6.0","cpu":{"mode":"custom","model":"qemu64","topology":{}},"displays":[],"interfaces":[{"type":"network","managed":null,"name":null,"mac":"52:54:00:57:46:03","model":"virtio","state":"up","source":{"bridge":null,"network":"default","portgroup":null,"dev":null,"mode":null,"address":null,"port":null,"local":{}},"address":{"bus":"0x00","function":"0x0","slot":"0x02","domain":"0x0000"}}],"redirectedDevices":[],"hostDevices":[],"filesystems":[],"watchdog":{},"vsock":{"cid":{}},"metadata":{"hasInstallPhase":false,"installSourceType":null,"installSource":null,"osVariant":null,"rootPassword":null,"userLogin":null,"userPassword":null}},"connectionName":"system","id":"/org/libvirt/QEMU/domain/_30100555_923e_4732_aae2_e776eb86d7c9","name":"subVmTest1","persistent":true,"autostart":false,"ui":{"initiallyExpanded":false,"initiallyOpenedConsoleTab":false}}

-> ph_mouse(".pf-v5-c-modal-box__footer button:contains(Remove):not([disabled]):not([aria-disabled=true])","click",0,0,0,false,false,false,false)
<- {'type': 'undefined'}
-> wait: !ph_is_present(".pf-v5-c-modal-box")
> debug: signal on /org/libvirt/QEMU: org.libvirt.Connect.DomainEvent(["/org/libvirt/QEMU/domain/_30100555_923e_4732_aae2_e776eb86d7c9",0,1])

> debug: subVmTest1.GET_VM(/org/libvirt/QEMU/domain/_30100555_923e_4732_aae2_e776eb86d7c9, system): update props {"inactiveXML":{"connectionName":"system","name":"subVmTest1","id":"/org/libvirt/QEMU/domain/_30100555_923e_4732_aae2_e776eb86d7c9","osType":null,"osBoot":[{"order":1,"type":"disk"},{"order":2,"type":"network"}],"firmware":null,"arch":"x86_64","currentMemory":131072,"memory":131072,"memoryBacking":false,"vcpus":{"count":"1","placement":"static","max":"1"},"disks":{"vda":{"target":"vda","driver":{"name":"qemu","type":"qcow2","cache":null,"discard":null,"io":null,"errorPolicy":null},"type":"file","device":"disk","source":{"file":"/var/lib/libvirt/images/subVmTest1-2.img","dev":null,"pool":null,"volume":null,"protocol":null,"name":null,"host":{},"startupPolicy":null},"bus":"virtio","readonly":false,"shareable":false,"removable":null}},"emulatedMachine":"pc-i440fx-rhel7.6.0","cpu":{"mode":"custom","model":"qemu64","topology":{}},"displays":[],"interfaces":[],"redirectedDevices":[],"hostDevices":[],"filesystems":[],"watchdog":{},"vsock":{"cid":{}},"metadata":{"hasInstallPhase":false,"installSourceType":null,"installSource":null,"osVariant":null,"rootPassword":null,"userLogin":null,"userPassword":null}},"connectionName":"system","id":"/org/libvirt/QEMU/domain/_30100555_923e_4732_aae2_e776eb86d7c9","name":"subVmTest1","persistent":true,"autostart":false,"ui":{"initiallyExpanded":false,"initiallyOpenedConsoleTab":false}}

> debug: signal on /org/libvirt/QEMU/domain/_30100555_923e_4732_aae2_e776eb86d7c9: org.libvirt.Domain.DeviceRemoved(["net0"])
```

That signal [decodes](https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainEventDefinedDetailType) to `VIR_DOMAIN_EVENT_DEFINED_UPDATED`. And indeed the next GET_VM call after that changes `interfaces:` to `[]`. The `DeviceRemoved net0` also confirms this.

The [first ampliefied debug-enabled run](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1289-20231106-074725-b1dd1907-rhel-9-4/log.html) was unfortunately successful, trying again.

**Comment by martinpitt** [comment]
Lots of failures on [rhel-9-4](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1289-20231106-135743-8124716b-rhel-9-4/log.html#80) and also [debian-testing](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1289-20231106-135743-8124716b-debian-testing/log.html). They show that the remove call happens, and succeeds:

```
> debug: detaching iface {"type":"number","value":0,"description":"0"} : ["virt-xml","-c","qemu:///system","subVmTest1","--remove-device","--network","1","--update"] {"err":"message","superuser":"try"}
```
and yet it doesn't, `domiflist` still shows it. Maybe the given index is wrong? Adding dumpxml debugging, and I also try to remove the interface using the CLI call directly instead of via the UI.

But one thing that I noticed is that the CLI call takes very long, about 10s. Perhaps it's just too slow? Testing that as the next step.

**Comment by martinpitt** [comment]
I tried to reproduce this outside of the test, with a pure CLI:
```
# virt-install --memory 50 --pxe --virt-type qemu --os-variant alpinelinux3.8 --disk none --wait 0 --name test1
# virsh domiflist test1
 Interface   Type      Source    Model   MAC
------------------------------------------------------------
 vnet0       network   default   e1000   52:54:00:8e:44:14

# virt-xml test1 --remove-device --network 1 --update --print-diff
--- Original XML
+++ Altered XML
@@ -66,14 +66,6 @@
       <alias name="virtio-serial0"/>
       <address type="pci" domain="0x0000" bus="0x00" slot="0x06" function="0x0"/>
     </controller>
-    <interface type="network">
-      <mac address="52:54:00:fd:65:7d"/>
-      <source network="default" portid="e34bacdb-9d52-449c-8e84-f9948c2b1664" bridge="virbr0"/>
-      <target dev="vnet1"/>
-      <model type="e1000"/>
-      <alias name="net0"/>
-      <address type="pci" domain="0x0000" bus="0x00" slot="0x03" function="0x0"/>
-    </interface>
     <serial type="pty">
       <source path="/dev/pts/1"/>
       <target type="isa-serial" port="0">

Device hotunplug successful.
```
but afterwards, `domiflist` still shows the same, and
```
# virsh dumpxml test1|grep -A3  '<interface'
    <interface type='network'>
      <mac address='52:54:00:fd:65:7d'/>
      <source network='default' portid='e34bacdb-9d52-449c-8e84-f9948c2b1664' bridge='virbr0'/>
      <target dev='vnet1'/>

# virsh dumpxml --inactive test1|grep -A3  '<interface'
    <interface type='network'>
      <mac address='52:54:00:fd:65:7d'/>
      <source network='default'/>
      <model type='e1000'/>
```

This is 100% reliably failing. I can even run the virt-xml command multiple times, and it keeps claiming success. I have a gut feeling that this requires cooperation from the guest OS? Obviously there's none, this is a forever-hanging PXE boot. That might explain the `self.waitCirrOSBooted(args['logfile'])`.

It works better with
```
virt-install --memory 128 --virt-type qemu --os-variant cirros0.3.0 --import --disk /var/lib/libvirt/images/cirros.img --wait 0 --name test1
```
Then the removal command works. I'll see if I can make it flaky that way.

**Comment by martinpitt** [comment]
[another failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1289-20231107-100846-a0b682ae-fedora-39-firefox/log.html#89), but now that I know how slow that is, I'm just going to bump the timeout.

File: src/helpers.js [hunk 1]
```diff
@@ -176,7 +176,7 @@ export function arrayEquals(arr1, arr2) {
 }
 
 export function logDebug() {
-    if (window.debugging === "all" || window.debugging === "machines")
+    if (window.debugging === "all" || window.debugging?.includes("machines"))
```
Comment (cockpituous) [inline src/helpers.js:179]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1289-20231107-105633-1b57fd30-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]

**Pull request merged**